build.sbt

```diff
@@ -36,17 +41,125 @@ lazy val cli = project
     },
     scriptedBufferLog := false,
     scriptedDependencies := {
       scriptedDependencies
-        .dependsOn(assembly)
+        .dependsOn(assembly, cliPack)
         .value
-    }
+    },
+    cliPackSettings
   )
 
+lazy val cliPackSettings = Def.settings(
+  cliPackLibJars := {
+    val s = streams.value
+    val log = s.log
+
+    val scalaNativeOrg = "org.scala-native"
+    val scalaBinVer = scalaBinaryVersion.value
+    val snVer = scalaNativeVersion.value
+
+    val scalaFullVers = crossScalaNative212
+    val cliAssemblyJar = assembly.value
+
+    // Standard modules needed for linking of Scala Native
+    val optLib = snVer match {
+      case "0.4.0" => Nil
+      case v       => "windowslib" :: Nil
+    }
+    val stdLibModuleIDs = Seq(
+      "nativelib",
+      "clib",
+      "posixlib",
+      "javalib",
+      "auxlib",
+      "scalalib"
+    ).++(optLib).map { lib =>
+      val nativeBinVersion = ScalaNativeCrossVersion.binaryVersion(snVer)
+      scalaNativeOrg % s"${lib}_native${nativeBinVersion}_${scalaBinVer}" % snVer
+    }
+    val compilerPluginModuleIDs =
+      scalaFullVers.map(v => scalaNativeOrg % s"nscplugin_$v" % snVer)
+    val allModuleIDs = (stdLibModuleIDs ++ compilerPluginModuleIDs).toVector
+    val allModuleIDsIntransitive = allModuleIDs.map(_.intransitive())
+
+    val resolvedLibJars = {
+      val retrieveDir = s.cacheDirectory / "cli-lib-jars"
+      val lm = {
+        import sbt.librarymanagement.ivy._
+        val ivyConfig = InlineIvyConfiguration().withLog(log)
+        IvyDependencyResolution(ivyConfig)
+      }
+      val dummyModuleName =
+        s"clilibjars-$snVer-$scalaBinVer-" + scalaFullVers.mkString("-")
+      val dummyModuleID = scalaNativeOrg % dummyModuleName % version.value
+      val descriptor =
+        lm.moduleDescriptor(
+          dummyModuleID,
+          allModuleIDsIntransitive,
+          scalaModuleInfo = None
+        )
+      lm
+        .retrieve(descriptor, retrieveDir, log)
+        .fold(
+          { unresolvedWarn => throw unresolvedWarn.resolveException },
+          identity
+        )
+        .distinct
+    }
+
+    cliAssemblyJar +: resolvedLibJars
+  },
+  cliPack / target := crossTarget.value / "pack",
+  cliPack / moduleName :=
+    s"scala-native-cli_${scalaBinaryVersion.value}-${scalaNativeVersion.value}",
+  cliPack / crossTarget := (cliPack / target).value / (cliPack / moduleName).value,
+  cliPack := {
+    val scalaBinVer = scalaBinaryVersion.value
+    val snVer = scalaNativeVersion.value
+
+    val trg = (cliPack / crossTarget).value
+    val trgLib = trg / "lib"
+    val trgBin = trg / "bin"
+
+    if (trg.exists)
+      IO.delete(trg)
+
+    IO.createDirectory(trgLib)
+    val libJars = cliPackLibJars.value
+    for (libJar <- libJars) {
+      IO.copyFile(libJar, trgLib / libJar.getName)
+    }
+
+    IO.createDirectory(trgBin)
+    val scriptDir = (Compile / sourceDirectory).value.getParentFile / "script"
+    println(scriptDir)
```

Comment:
Is this on purpose? I know I leave a lot of println's by mistake

Reply:
Thank you for spotting this one, removed!

Reply:
And when it comes to `@ProgName` annotation we might think about it. If we would like to add more cli programs, like for example `scalap` counterpart, then it might be useful to rename our main class to `ScalaNativeLinker` or `ScalaNativeLd` as well. This would also align it with Scala.js-cli naming convention